ang/crmCaseType.js

```diff
@@ -263,33 +261,54 @@
       $scope.activityTypes = _.indexBy(apiCalls.actTypes.values, 'name');
       $scope.activityTypeOptions = _.map(apiCalls.actTypes.values, formatActivityTypeOption);
       $scope.defaultAssigneeTypes = apiCalls.defaultAssigneeTypes.values;
-      $scope.relationshipTypeOptions = _.map(apiCalls.relTypes.values, function(type) {
-        return {id: type[REL_TYPE_CNAME], text: type.label_b_a};
-      });
-      $scope.defaultRelationshipTypeOptions = getDefaultRelationshipTypeOptions();
+      $scope.relationshipTypeOptions = getRelationshipTypeOptions(false);
+      $scope.defaultRelationshipTypeOptions = getRelationshipTypeOptions(true);
       // stores the default assignee values indexed by their option name:
       $scope.defaultAssigneeTypeValues = _.chain($scope.defaultAssigneeTypes)
         .indexBy('name').mapValues('value').value();
     }
 
-    /// Returns the default relationship type options. If the relationship is
-    /// bidirectional (Ex: Spouse of) it adds a single option otherwise it adds
-    /// two options representing the relationship type directions
-    /// (Ex: Employee of, Employer is)
-    function getDefaultRelationshipTypeOptions() {
+    // Returns the relationship type options. If the relationship is
+    // bidirectional (Ex: Spouse of) it adds a single option otherwise it adds
+    // two options representing the relationship type directions (Ex: Employee
+    // of, Employer of).
+    //
+    // The default relationship field needs values that are IDs with direction,
+    // while the role field needs values that are names (with implicit
+    // direction).
+    //
+    // At any rate, the labels should follow the convention in the UI of
+    // describing case roles from the perspective of the client, while the
+    // values must follow the convention in the XML of describing case roles
+    // from the perspective of the non-client.
+    function getRelationshipTypeOptions($isDefault) {
       return _.transform(apiCalls.relTypes.values, function(result, relType) {
         var isBidirectionalRelationship = relType.label_a_b === relType.label_b_a;
-
-        result.push({
-          label: relType.label_b_a,
-          value: relType.id + '_b_a'
-        });
-
-        if (!isBidirectionalRelationship) {
+        if ($isDefault) {
           result.push({
-            label: relType.label_a_b,
+            label: relType.label_b_a,
             value: relType.id + '_a_b'
           });
+
+          if (!isBidirectionalRelationship) {
+            result.push({
+              label: relType.label_a_b,
+              value: relType.id + '_b_a'
+            });
+          }
+        }
+        else {
+          result.push({
+            text: relType.label_b_a,
+            id: relType.name_a_b
```

Comment:
@demeritcowboy I just confirmed what you found.  This here should be `relType.label_a_b` rather than `relType.name_a_b` and the same thing six lines below with the `b_a` equivalent.

Case types define roles by the *labels*, not names.  That's really bad, from my perspective, but out of scope for us to address here.  Addressing that would also require accommodating legacy XML files.

You found this because creating new relationship types in the UI will munge the relationship type name, so the name of a multi-word releationship type will never be the same as the label.  This is new from last year post #12097--it used to be identical to the label on a single-lingual site.

Anyway, @alifrumin is out for the afternoon, but I'm sure that she can fix it to use the label tomorrow.  You might experiment a bit locally with these two line changes.

I'd also be interested in your thoughts about whether to switch back to not munging the relationship type label.  That's still the case in CRM_Contact_BAO_RelationshipType::add()
https://github.com/civicrm/civicrm-core/blob/65e7186ec68f7f26abb20f19145696af2012e14f/CRM/Contact/BAO/RelationshipType.php#L94-L100

But the relationship type form no longer uses that function and instead munges it before sending it to the RelationshipType.create API:
https://github.com/civicrm/civicrm-core/blob/65e7186ec68f7f26abb20f19145696af2012e14f/CRM/Admin/Form/RelationshipType.php#L209-L213

I don't know if @eileenmcnaughton had a specific reason for doing this, but I wonder if going back to name = label might reduce the pain of switching case types to defining roles with names later.

Reply:
@agh1 I can't recall one. My best guess was just to make it more like a machine name - perhaps copying a convention on another field somewhere but it seems like a 'flip' decision rather than something done for an important reason so I don't see any reason not to change it

Reply:
The xml definition using label isn't how it was originally designed. I agree it's out of scope here but for reference (https://lab.civicrm.org/dev/core/issues/774#note_14494). It got messed up a while ago and then kept getting further entangled. I still have some worries about the removal of the REL_TYPE_CNAME and its comment in this PR since at least that's currently there as a reminder.

Regarding the munging, I also don't immediately see any reason it needs to be that way, unless it's maybe some kind of naming requirement for the mentioned custom metadata where spaces or whatever might cause trouble. Does that metadata get output into html attributes or something?

Another thought while I'm here which is maybe out of scope but could affect several people: Should probably check that an existing install with civicrm_webform case integration would still do what it did before.